modules/sor/sorV2/lib/router.ts

```diff
@@ -75,34 +76,140 @@ export class Router {
 
         const orderedQuotePaths = valueArr.map((item) => item.item);
 
-        // If there is only one path, return it
-        if (orderedQuotePaths.length === 1) {
-            return orderedQuotePaths;
-        }
-
         // Split swapAmount in half, making sure not to lose dust
-        const swapAmount50up = swapAmount.mulDownFixed(WAD / 2n);
-        const swapAmount50down = swapAmount.sub(swapAmount50up);
-
-        const path50up = new PathWithAmount(orderedQuotePaths[0].tokens, orderedQuotePaths[0].pools, swapAmount50up);
-        const path50down = new PathWithAmount(
-            orderedQuotePaths[1].tokens,
-            orderedQuotePaths[1].pools,
-            swapAmount50down,
+        const bestPath = orderedQuotePaths[0];
+        // console.log('bestPath pools', bestPath.pools);
+
+        const bestPathNormalizedLiquidities = bestPath.pools.map((p, i) =>
+            p.getNormalizedLiquidity(bestPath.tokens[i], bestPath.tokens[i + 1]),
         );
+        // console.log('bestPathNormalizedLiquidities', bestPathNormalizedLiquidities);
 
-        if (swapKind === SwapKind.GivenIn) {
-            if (
-                orderedQuotePaths[0].outputAmount.amount >
-                path50up.outputAmount.amount + path50down.outputAmount.amount
-            ) {
-                return orderedQuotePaths.slice(0, 1);
+        const bestNormLiq = MathSol.powDownFixed(
+            bestPathNormalizedLiquidities.reduce((acc, normLiq) => MathSol.mulDownFixed(acc, normLiq), WAD),
+            WAD / BigInt(bestPathNormalizedLiquidities.length),
+        );
+        // const bestNormLiq = bestPathNormalizedLiquidities.reduce(
+        //     (acc, normLiq) => MathSol.min(acc, normLiq),
+        //     maxInt256,
+        // );
+        // console.log('bestNormLiq  ', bestNormLiq);
```

Comment:
did you try with the max normLiq? Or would that usually yield the same normLiq for both bestPath and 2ndBestPath?

Reply:
No, I didn't try with max normLiq, but I don't think that makes much sense. 
The pool with the lowest normalized liquidity is the one that will have most impact on the swap path. 
In other words, even if a path had many pools with infinite liquidity, a single pool with low liquidity acts as a "bottleneck" to the whole path.
That's why a good approximation is to use the minNormLiq of the path as a representation of the whole path liquidity. Then, you can compare paths with that amount.
To me, the geometric mean is the best representation, but minNormLiq should be good enough.
Anyway, none of them achieve the result I expected, so not really sure anymore 😅

Reply:
I'm not sure I agree. Let's assume two paths, where one has 3 hops with 1000 normLiq and 1 hop with 100 normLiq. The other path has 4 hops with 500 normLiq. If that one 100 normLiq path offers a good swap (maybe even a positive arb) the first path is probably better as most of the hops offer better normLiq. Just a thought and might be worth wile a test.

Reply:
Think the other way around. Low normLiq means high price impact. So let's say you have a few pools with 0.1% priceImpact and a single one with 10%. Your result will be worse than 10% priceImpact no matter how many high liq pools that path has. Another path with ~1% priceImpact on each pool should have better results.
In any case, it should be straightforward to setup a test an validate the theory. I'll do it over here.